### PR TITLE
SQS Long Polling plus a few other tweaks

### PR DESCRIPTION
**Pull request by dwoo4dwoo**
- added SQS Long Polling (enabled by default with wait_time_seconds = 10s)

http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-long-polling.html

- reduced receive_interval default to 0.1 given the above (maybe 0 would
  be better?)

- documented max_number_of_messages in README.rdoc

- increased max_number_of_messages default to 10 to match out_sqs.rb

### CODE REVIEW

File: lib/fluent/plugin/in_sqs.rb [hunk 1]
```diff
@@ -14,8 +14,9 @@ def initialize
     config_param :tag, :string
     config_param :sqs_endpoint, :string, :default => 'sqs.ap-northeast-1.amazonaws.com'
     config_param :sqs_url, :string
-    config_param :receive_interval, :time, :default => 1
-    config_param :max_number_of_messages, :integer, :default => 1
+    config_param :receive_interval, :time, :default => 0.1
```
Comment (ixixi) [inline lib/fluent/plugin/in_sqs.rb]: change `receive_interval` value  in specfile.
https://github.com/ixixi/fluent-plugin-sqs/blob/master/spec/lib/fluent/plugin/in_sqs_spec.rb#L44
```
    context do
      subject {instance.receive_interval}
      it{should == 1}
    end
```

File: lib/fluent/plugin/in_sqs.rb [hunk 2]
```diff
@@ -47,7 +48,10 @@ def run_periodic
       until @finished
         begin
           sleep @receive_interval
-          @queue.receive_message(:limit => @max_number_of_messages) do |message|
+          @queue.receive_message(
+            :limit => @max_number_of_messages,
+            :wait_time_seconds => @wait_time_seconds
```
Comment (ixixi) [inline lib/fluent/plugin/in_sqs.rb]: https://github.com/ixixi/fluent-plugin-sqs/blob/master/spec/lib/fluent/plugin/in_sqs_spec.rb#L79
``` diff
@@ -82,7 +82,7 @@ describe do
             })
         end
       end
-      expect_any_instance_of(AWS::SQS::Queue).to receive(:receive_message).with({:limit => 10}).at_least(:once).and_call_original
+      expect_any_instance_of(AWS::SQS::Queue).to receive(:receive_message).with({:limit => 10, :wait_time_seconds=>10}).at_least(:once).and_call_original

       d = driver
       d.run do
```

**Comment by dwoo4dwoo** [comment]
Hello,

Thank you for catching the spec file changes I missed.  Sorry about that!  As you can see ... I've pushed another commit to resolve the missing items.

David

**Comment by ixixi** [comment]
Sorry for the delay.
released v1.6.2

**Pull request merged**